File: Leetcode/161/161.go
func f(long *string, short *string) bool {
	var i = 0
	for i = 0; i < len(*short); i++ {
		if (*long)[i] != (*short)[i] {
			break
		}
	}
	i++
	for ; i < len(*long); i++ {
		if (*long)[i] != (*short)[i-1] {
			return false
		}
	}
	return true
}
func isOneEditDistance(s string, t string) bool {
	if len(s) == len(t) {
		diff := 0
		for i := 0; i < len(s); i++ {
			if s[i] != t[i] {
				diff++
				if diff > 1 {
					break
				}
			}
		}
		return diff == 1
	} else if len(s) - len(t) == 1 {
		return f(&s, &t)
	} else if len(t) - len(s) == 1 {
		return f(&t, &s)
	}
	return false
}

File: Leetcode/161/161.cpp
class Solution {
public:
    bool f(string &a, string &b) {
        int lena = (int)a.size();
        int lenb = (int)b.size();
        int i;
        for (i = 0; i < lenb; ++i) {
            if (a[i] != b[i]) {
                break;
            }
        }
        ++i;
        for (; i < lena; ++i) {
            if (a[i] != b[i-1]) {
                return false;
            }
        }
        return true;
    }
    bool isOneEditDistance(string s, string t) {
        int n = (int)s.size();
        int m = (int)t.size();
        if (n == m) {
            int diff = 0;
            for (int i = 0; i < n; ++i) {
                if (s[i] != t[i]) {
                    ++diff;
                    if (diff > 1) {
                        break;
                    }
                }
            }
            return diff == 1;
        }
        else if (m == n - 1) {
            return f(s, t);
        }
        else if (n == m - 1) {
            return f(t, s);
        }
        return false;
    }
};